src/main/java/mate.academy/lib/Injector.java

```diff
@@ -47,20 +47,20 @@ private Object createNewInstance(Class<?> clazz) {
             return instances.get(clazz);
         }
         try {
-            if (clazz.isAnnotationPresent(Component.class)) {
-                Constructor<?> constructor = clazz.getConstructor();
-                Object instance = constructor.newInstance();
-                instances.put(clazz, instance);
-                return instance;
-            }
-            throw new RuntimeException("Can't create a new instance of " + clazz.getName()
-                    + ". It doesn't contain @Component annotation");
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
         } catch (ReflectiveOperationException e) {
             throw new RuntimeException("Can't create a new instance of " + clazz.getName());
         }
     }
 
     private Class<?> findImplementation(Class<?> interfaceClazz) {
+        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
```

Comment:
please read carefully your code here and run `main()` method

you should check the implementation, not an interface, for annotation
and you shouldn't check anything if `interfaceClazz` is a simple class which does not implement any interface